src/stdlib/SDL_string.c

```diff
@@ -307,6 +307,17 @@ int
 SDL_memcmp(const void *s1, const void *s2, size_t len)
 {
 #if defined(HAVE_MEMCMP)
+#if defined(__vita__)
+    /*
+      Using memcmp on NULL is UB per POSIX / C99 7.21.1/2.
+      But, both linux and bsd allow that, with an exception:
+      zero length strings are always identical, so NULLs are never dereferenced.
+      sceClibMemcmp on PSVita doesn't allow that, so we check ourselves.
+    */
+    if (len == 0) {
+      return 0;
+    }
+#endif
```

Comment:
It might be cleaner to call `sceClibMemcmp` directly here rather than using defines.

```suggestion
#if defined(__vita__)
    /*
      Using memcmp on NULL is UB per POSIX / C99 7.21.1/2.
      But, both linux and bsd allow that, with an exception:
      zero length strings are always identical, so NULLs are never dereferenced.
      sceClibMemcmp on PSVita doesn't allow that, so we check ourselves.
    */
    if (len == 0) {
      return 0;
    }
    return sceClibMemcmp(s1, s2, len);
#elif defined(HAVE_MEMCMP)
```